src/Symfony/Component/Security/Http/Authenticator/AbstractPreAuthenticatedAuthenticator.php

```diff
@@ -79,6 +79,17 @@ public function supports(Request $request): ?bool
             return false;
         }
 
+        // do not overwrite already stored tokens from the same user (i.e. from the session)
+        $token = $this->tokenStorage->getToken();
+
+        if (null !== $token && $token->getUserIdentifier() === $username) {
```

Comment:
What about reproducing the check in the legacy security system exactly?

https://github.com/symfony/symfony/blob/c143be5b2c8c91e139abfd086f482cea18c0c896/src/Symfony/Component/Security/Http/Firewall/AbstractPreAuthenticatedListener.php#L89-L94

```suggestion
        if ($token instanceof PreAuthenticatedToken && $this->firewallName === $token->getFirewallName() && $token->getUserIdentifier() === $username) {
```

Reply:
Yes, this looks more solid. I can confirm that using this check the Authenticator still works as expected.